tests/functional/test_exclude_newer.py

```diff
@@ -0,0 +1,93 @@
+"""Tests for pip install --exclude-newer-than."""
+
+from __future__ import annotations
+
+import pytest
+
+from tests.lib import PipTestEnvironment, TestData
+
+
+class TestExcludeNewer:
+    """Test --exclude-newer-than functionality."""
+
+    def test_exclude_newer_than_invalid_date(
+        self, script: PipTestEnvironment, data: TestData
+    ) -> None:
+        """Test that --exclude-newer-than fails with invalid date format."""
+        result = script.pip(
+            "install",
+            "--no-index",
+            "-f",
+            data.packages,
+            "--exclude-newer-than=invalid-date",
+            "simple",
+            expect_error=True,
+        )
```

Comment:
Pro-tip: you can use `script.pip_install_local()` when you don't want to access the Internet.

```suggestion
        result = script.pip_install_local(
            "--exclude-newer-than=invalid-date", "simple", expect_error=True
        )
```